File: fact.rb
#!/usr/bin/env ruby

# @@@ get_version: ruby --version | awk '{print $2}' | perl -pe 's/p\d+$//'

times = 5000000
fact_16 = 20922789888000

def fact(n)
    if n == 1
        return 1
    end
    return n * fact(n - 1)
end

ok = true
for i in 1 .. times
    ok = ok && fact(16) == fact_16
end

puts "ruby finish #{times} - #{ok ? 'ok' : 'fail'}"

File: fact.dart
#!/usr/bin/env dart

// @@@ get_version: dart --version 2>&1 | awk '{print $4}' | perl -pe 's/_.+$//'
// @@@ is_fast: 1

const times = 200000000;
const fact_16 = 20922789888000;

int fact(int n) {
    if (n == 1) {
        return 1;
    }
    return n * fact(n - 1);
}

void main() {
    bool ok = true;
    for (int i = 0; i < times; i++) {
        ok = ok && fact(16) == fact_16;
    }

    print("Dart finish ${times.toString()} - ${ok ? 'ok' : 'fail'}");
}
